addons/survey/static/src/js/survey_form.js

```diff
@@ -96,9 +100,19 @@ publicWidget.registry.SurveyFormWidget = publicWidget.Widget.extend({
             var $form = this.$('form');
             var formData = new FormData($form[0]);
 
-            this._prepareSubmitValues(formData, params);
+            if ($target.hasClass('o_survey_timer')) {
+                $target.val('finish');
+            } else {
+                // Validation pre submit
+                this._resetErrors();
```

Comment:
"resetErrors" and "showErrors" could be made in the "_validateForm" method?
And it would return a false value if we find any validation error so you know you have to "return" here?

Would be cleaner IMO.